test/e2e/utils.js

```diff
@@ -221,6 +253,36 @@ const getServiceEnv = async (file, service, envVarName) => {
   return lines[2];
 };
 
+const hasBackupDir = () => fs.existsSync(backupFolder);
+
+/**
+ * Reads the backup directory recursively for service image versions
+ * @param {string} filename 
+ * @returns {Object.<string, string>} Dictionary with {service: version}. E.g. {"one": "1.0.0"}
+ */
+const readBackupVersions = async (filename) => {
+  const versions = {};
+
+  const backupFolders = await fs.promises.readdir(backupFolder);
+  const lastBackupFolder = backupFolders.at(-1);
+  const dir = await fs.promises.readdir(path.resolve(backupFolder, lastBackupFolder));
+  for (const filePath of dir) {
+    if (filePath === filename) {
+      const fileContent = await fs.promises.readFile(
+        path.resolve(backupFolder, lastBackupFolder, filePath),
+        { 'encoding': 'utf-8' }
+      );
+      for (const line of fileContent.split('\n')) {
+        const matched = line.match(/image: \S+\/(?<service>[^:/]+):(?<version>\d\.\d\.\d)$/);
```

Comment:
I wonder if you could just compare it with a file that we generate. I was thinking to using `setVersion` function from this file to return a string that we compare with entirely, instead of parsing lines.

Reply:
Added in 994f6b8